### DISCUSSION TITLE
Remove ActivityPropagationHandler breaking change

### DISCUSSION THREAD

**Discussion by leastprivilege** [dotnet/yarp#1316] [Q&A] [answered]
We have been using the activity handler with the IHttpForwarder.

Could you elaborate how to deal with this in RC.1 for both .NET 6 and earlier?

thanks!

**Comment by MihaZupan** [comment, marked as answer]
Hi,

We originally added the handler to YARP as a workaround to bridge the [gap in `SocketsHttpHandler`, where it couldn't make use of distributed tracing out of the box](https://github.com/dotnet/runtime/issues/31261).
The good news is that we made a lot of improvements in the area for the 6.0 release with the introduction of the [`DistributedContextPropagator`](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.distributedcontextpropagator) class. As part of the change, we enabled distributed tracing on `SocketsHttpHandler` by default as well, including the option to configure it via the [`ActivityHeadersPropagator`](https://github.com/dotnet/runtime/issues/55556) property.

**As such, the scenario is expected to "just work" in YARP on .NET 6.0 and newer by default, there is no need to use workarounds like the `ActivityPropagationHandler`.**

For versions before 6.0, YARP will work as a silent pass-through for distributed tracing by default - the trace will not be broken, but the proxy will not participate in it either.
See [updated docs](https://github.com/microsoft/reverse-proxy/blob/main/docs/docfx/articles/distributed-tracing.md) for getting YARP to actively participate.

Can you also describe how you are currently using the handler (just setting the config flag, wrapping the handler manually, using vendor-specific distributed tracing packages ...)?

**Reply by karelz** [reply]
It may be worth saying that we plan to document the above workaround for YARP active participation in distributed tracing on .NET 5/.NET Core 3.1 in YARP docs.

**Reply by leastprivilege** [reply]
Thanks @MihaZupan for the explanation.

We will wire up the handler manually for non-NET6 scenarios.

> Can you also describe how you are currently using the handler 

We are using it internally for forward local API calls to remote APIs in our BFF product
https://github.com/DuendeSoftware/BFF/blob/6dde5ccf527121a275d48a79262f46bff6c59c4f/src/Duende.Bff.Yarp/RemoteApiEndpoint.cs#L74

**Reply by MihaZupan** [reply]
FYI we added more documentation with a more capable workaround here: https://github.com/microsoft/reverse-proxy/blob/main/docs/docfx/articles/distributed-tracing.md
